client/app/components/navbar/navbar.service.js

```diff
@@ -0,0 +1,24 @@
+import template from './navbar-menu-header/navbar-menu-header.component.html';
+
+{
+  class Navbar {
+    constructor($compile, $rootScope, $timeout) {
+      this.$compile = $compile;
+      this.$rootScope = $rootScope;
+      this.$timeout = $timeout;
+    }
+
+    buildMenuHeader(content) {
+      const compiledTemplate = this.$compile(template)(_.merge(
+        this.$rootScope.$new(true),
+        { $ctrl: { content } },
+      ));
+      // $timeout is required in order to let angular's scope $digest
+      return this.$timeout(() => compiledTemplate.html());
```

Comment:
not mandatory but a sighly better way : 
$timeout will do a digest for every single scope starting from the rootScope
you only need to digest the scope used for compilation
```
$compile(myTemplate)(myScope);
myScope.$digest();
```